dlt/destinations/snowflake/snowflake.py

```diff
@@ -48,44 +53,82 @@
     "VARIANT": "complex"
 }
 
+@with_config(spec=AwsCredentials, sections=(known_sections.DESTINATION, "filesystem", "credentials"))
+def _s3_config(config: AwsCredentials = config.value) -> AwsCredentials:
+    return config
+
+
 class SnowflakeLoadJob(LoadJob, FollowupJob):
     def __init__(
             self, file_path: str, table_name: str, write_disposition: TWriteDisposition, load_id: str, client: SnowflakeSqlClient,
-            stage_name: Optional[str] = None, keep_staged_files: bool = True
+            stage_name: Optional[str] = None, storage_integration: Optional[str] = None, keep_staged_files: bool = True
     ) -> None:
         file_name = FileStorage.get_file_name_from_file_path(file_path)
         super().__init__(file_name)
 
-        with client.with_staging_dataset(write_disposition == "merge"):
+        with client.with_staging_dataset(write_disposition in ["merge", "replace"]):
             qualified_table_name = client.make_qualified_table_name(table_name)
 
-            if stage_name:
+            # extract and prepare some vars
+            bucket_path = CopyFileLoadJob.get_bucket_path(file_path) if CopyFileLoadJob.is_reference_job(file_path) else ""
+            file_name = FileStorage.get_file_name_from_file_path(bucket_path) if bucket_path else file_name
+            from_clause = ""
+            credentials_clause = ""
+            pattern_clause = ""
+            stage_file_path = ""
+
+            # create storage integration stage for reference jobs if defined, for now gcs only
+            if bucket_path and storage_integration:
+                stage_name = client.make_qualified_table_name(stage_name) if stage_name else qualified_table_name
+                bucket = CopyFileLoadJob.get_bucket(file_path)
+                client.execute_sql(f"""
+                    CREATE STAGE IF NOT EXISTS {stage_name}
+                    URL = '{bucket}'
+                    STORAGE_INTEGRATION = {storage_integration};""")
+                from_clause = f"FROM @{stage_name}"
+                # select all files that match the file from our job
+                pattern_clause = f"PATTERN = '.*{file_name}'"
```

Comment:
actually, we do we need pattern? we can just give a file name in FILES part of the COPY command